internal/re2/re2.go

```diff
@@ -12,11 +12,20 @@ import (
 )
 
 //export cre2_new
-func cre2New(patternPtr unsafe.Pointer, patternLen uint32, flags uint32) unsafe.Pointer
+func cre2New(patternPtr unsafe.Pointer, patternLen uint32, opts unsafe.Pointer) unsafe.Pointer
 
 //export cre2_delete
 func cre2Delete(rePtr unsafe.Pointer)
 
+//export cre2_opt_new
+func cre2OptNew() unsafe.Pointer
+
+//export cre2_opt_delete
+func cre2OptDelete(ptr unsafe.Pointer)
+
+//export cre2_opt_set_max_mem
+func cre2OptSetMaxMem(ptr unsafe.Pointer, maxMem uint64)
```

Comment:
Very simple nit: Shall we specify the units on this? like maxMemInBytes?

Reply:
While it's not required I think I'd rather have the function name match the imported name as convention at least for now